src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,135 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private Object[] elementList;
+    private int size;
+
+    public ArrayList() {
+        this.elementList = EMPTY_ELEMENT_DATA;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementList.length) {
+            elementList = grow();
+        }
+        elementList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can not add element, "
+                    + "index is out of bounds.");
+        }
+        if (size == elementList.length) {
+            elementList = grow();
+        }
+        System.arraycopy(elementList, index, elementList, index + 1, size - index);
+        elementList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] listAsArray = list.toArray();
+        if (listAsArray.length > elementList.length - size) {
+            elementList = grow(listAsArray.length + size);
+        }
+        System.arraycopy(listAsArray, 0, elementList, size, listAsArray.length);
+        size += listAsArray.length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size - 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can not get element, "
+                    + "index is out of bounds.");
+        }
```

Comment:
```suggestion
        if (index >= size || index < 0) {
            throw new ArrayListIndexOutOfBoundsException("Can not get element, "
                    + "index is out of bounds.");
        }
```
make this a private utility method